src/components/OpenWith/OpenWith.js

```diff
@@ -192,9 +204,69 @@ class OpenWith extends PureComponent<Props, State> {
      * @private
      * @return {void}
      */
-    onIntegrationClick(): void {
-        /* no-op */
-    }
+    onIntegrationClick = (appIntegrationId: string): void => {
+        const { fileId }: Props = this.props;
+        this.api
+            .getAppIntegrationsAPI(false)
+            .execute(
+                appIntegrationId,
+                fileId,
+                this.executeIntegrationSuccessHandler,
+                this.executeIntegrationErrorHandler,
+            );
+    };
+
+    /**
+     * Opens the integration in a new tab based on the API data
+     *
+     * @private
+     * @param {ExecuteAPI} executePostData - API response on how to open an executed integration
+
+     * @return {void}
+     */
+    executeIntegrationSuccessHandler = (executePostData: ExecuteAPI): void => {
+        const { method, url } = executePostData;
+        switch (method) {
+            case HTTP_POST:
+                this.setState({ executePostData });
+                break;
+            case HTTP_GET:
+                // Use a unique combination of integration + file ID to avoid creation of new windows
+                const windowRef = window.open(url);
+
+                if (!windowRef) {
+                    this.executeIntegrationErrorHandler(
```

Comment:
Have you considered opening the integration in a new tab instead of a new window?

Reply:
I don't think I can control that from JS? Pal mentioned that there is a setting on their side that dictates. Currently some integrations open via a new tab using this method.

Reply:
Nevermind about this. How its executed is dependent on the users browser settings 👍